assets/dev/js/editor/data/globals/component.js

```diff
@@ -4,11 +4,16 @@ import ColorsComponent from './colors/component';
 
 import * as commandsData from './commands/data/';
 
+const ControlsCSSParser = require( 'elementor-editor-utils/controls-css-parser' );
+
 export default class Component extends ComponentBase {
 	__construct( args = {} ) {
 		super.__construct( args );
 
 		elementorCommon.elements.$window.on( 'elementor:loaded', this.onElementorLoaded.bind( this ) );
+
+		elementor.on( 'document:loaded', () => this.updateTempStylesheet() );
+		elementor.on( 'preview:loaded', () => this.updateTempStylesheet() );
```

Comment:
Do you really need to do it twice, or you may set a flag to indicate when both document and preview were loaded

Reply:
It's for two different situations. `document:loaded` is used in order to remove\render on switch between to kit and back.
the `preview:loaded` is used to re-render on switch to canvas and back.

Reply:
What will happen in the first loading? It seems like both of the events will fire and the CSS will be updated twice.